### PR TITLE
Support muted test

### PR DESCRIPTION
**Pull request by nprizal**
### Description

Test Engine has added `muted` state for the test. This PR adds support in `bktec` to handle muted tests.
When a test is muted in Test Engine, a failure in that test will be treated as if it was passed. 
Test Collectors will still report the failure to the Test Engine.

<table>
  <tr>
    <td>Screenshot 2024-11-07 at 3 44 52 PM</td>
    <td>Screenshot 2024-11-07 at 3 45 23 PM</td>
  </tr>
</table>

### Changes

- Refactor `Run` function for each runner to return failed tests as a `TestCases` struct instead of a `string` so the main function can check if the failed tests are muted or not based on its `Scope` and `Name`. The function also takes an array of `TestCases` instead of `string` to follow the changes.
- Fix Playwright result handling. Playwright use a nested structure in their JSON result. Previously, only tests that **aren't** nested get returned from `Run` function. With the changes, it recursively finds all failed tests.
- Update main function to check if failed tests are muted or not.
- Update the pact provider name following the product name changes

### CODE REVIEW

**Comment by wooly** [comment]
I paired with @nprizal on this and we worked through the logic together and it's working well. One can that we decided to kick down the road a bit was that I found it a little bit jarring that we failed the run inside the runner then marked the run as passed outside the runner logic.

For a first cut though, I think it's reasonable.

Nice work @nprizal!

**Review by wooly — APPROVED** [review]

**Review by niceking — APPROVED** [review]
Looks good, just a few non blocking comments about the information we present to the user

**Comment by niceking** [comment]
> One can that we decided to kick down the road a bit was that I found it a little bit jarring that we failed the run inside the runner then marked the run as passed outside the runner logic.

@wooly what did you find jarring about this? How did you expect this to be done?

File: main.go [hunk 1]
```diff
@@ -157,6 +152,40 @@ func runTestsWithRetry(testRunner TestRunner, testsCases *[]string, maxRetries i
...
+				scopeName := test.Scope + "/" + test.Name
+				if _, ok := mutedTestMap[scopeName]; !ok {
+					failedTests = append(failedTests, test)
+				} else {
+					mutedTestForRun = append(mutedTestForRun, test)
+				}
+			}
+
+			if len(mutedTestForRun) > 0 {
+				fmt.Println("⚠️ Following tests are muted and will not be retried or fail the run:")
```
[the remark this answers is not in the document]
Comment (nprizal) [reply main.go:176]: Ah yes, that's a good point. This is the same direction as displaying the summary at the end of the test. I'll make this change.

**Pull request merged**